core/templates/services/messenger.service.ts

```diff
@@ -19,147 +19,201 @@
  * be attempted due to cross-domain security issues.)
  */
 
-angular.module('oppia').factory('MessengerService', [
-  '$log', '$window', function($log, $window) {
-    var isPositiveInteger = function(n) {
-      return (typeof n === 'number' && n % 1 === 0 && n > 0);
-    };
-    var isBoolean = function(b) {
-      return typeof b === 'boolean';
-    };
-
-    var SUPPORTED_HASHDICT_VERSIONS = [
-      '0.0.0', '0.0.1', '0.0.2', '0.0.3'
-    ];
-
-    var MESSAGE_VALIDATORS = {
-      heightChange: function(payload) {
-        return isPositiveInteger(payload.height) && isBoolean(payload.scroll);
-      },
-      explorationLoaded: function() {
-        return true;
-      },
-      stateTransition: function(payload) {
-        return Boolean(payload.oldStateName) || Boolean(payload.newStateName);
-      },
-      explorationReset: function(payload) {
-        return Boolean(payload.stateName);
-      },
-      explorationCompleted: function() {
-        return true;
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+
+import { LoggerService } from 'services/contextual/logger.service';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface MessageValidatorsType {
+  heightChange: ((payload: {height: number; scroll: number}) => boolean);
```

Comment:
If not, please use a plain function signature instead.

```typescript
interface MessageValidatorsType {
  heightChange(...): boolean;
  // ...
}
```

Reply:
Did not understand this, why not complete type?

Reply:
Just for consistency. The old-code mixed the two types of function declarations:
```typescript
obj = { name: (args) => returnType };
// vs.
obj = { name(args): returnType };
```

My ask was to prefer the second declaration.

Reply:
Ok, so here is what I think:

The first method is only used when we do not define the function but just declare the type of the function and the second method is used when the function is being defined.

So, that is why I am using the first method here.

@bansalnitish can you please confirm?

Reply:
-   Both styles can be used in each of the scenarios you listed (declaration vs. definition), so it's still not clear to me why one is chosen  here over the other?
-   I asked for Style 2 because it's closer to how we define classes, and because there's less noise.
-   Style 1 is only "necessary" when defining an argument to a function. Otherwise, I (think) Style 2 can always be used. Seems we need an official "style guide" recommendation for choosing which to use.
    -   @bansalnitish thoughts?

Reply:
I agree with @brianrodri, both styles can be used in each of scenarios and we should use style 2 because there is less noise. So, I would say go ahead with style 2.

Reply:
Done.